model/src/main/java/oracle/kubernetes/weblogic/domain/v2/Domain.java

```diff
@@ -271,44 +277,18 @@ public V1SecretReference getAdminSecret() {
     return spec.getAdminSecret();
   }
 
-  /**
-   * Returns the name of the admin server.
-   *
-   * @return admin server name
-   */
-  public String getAsName() {
-    return spec.getAsName();
-  }
-
-  /**
-   * Returns the port used by the admin server.
-   *
-   * @return admin server port
-   */
-  public Integer getAsPort() {
-    return spec.getAsPort();
-  }
-
   /**
    * Returns the domain unique identifier.
    *
    * @return domain UID
    */
   public String getDomainUID() {
-    return spec.getDomainUID();
-  }
-
-  /**
-   * Returns the domain name
-   *
-   * @return domain name
-   */
-  public String getDomainName() {
-    return spec.getDomainName();
+    return spec.getDomainUID() != null ? spec.getDomainUID() : getMetadata().getName();
```

Comment:
In Java 8, use of Optional is preferred:

return Optional.ofNullable(spec.getDomainUID()).orElse(getMetadata().getName());

This removes the need to refer to spec.getDomainUID() twice, and ensures that the reader can tell it is the same value.